mycroft/skills/event_scheduler.py

```diff
@@ -281,20 +280,35 @@ def shutdown(self):
 class EventSchedulerInterface:
     """Interface for accessing the event scheduler over the message bus."""
 
-    def __init__(self, name, sched_id=None, bus=None):
-        self.name = name
-        self.sched_id = sched_id or self.__class__.__name__
+    def __init__(self, name=None, sched_id=None, bus=None, skill_id=None):
+        # NOTE: can not rename or move sched_id/name arguments to keep api compatibility
+        if name:
+            LOG.warning("name argument has been deprecated! use skill_id instead")
+        if sched_id:
+            LOG.warning("sched_id argument has been deprecated! use skill_id instead")
+
+        self.skill_id = skill_id or sched_id or name or self.__class__.__name__
         self.bus = bus
         self.events = EventContainer(bus)
-
         self.scheduled_repeats = []
 
     def set_bus(self, bus):
+        """Attach the messagebus of the parent skill
+
+        Args:
+            bus (MessageBusClient): websocket connection to the messagebus
+        """
         self.bus = bus
         self.events.set_bus(bus)
 
     def set_id(self, sched_id):
-        self.sched_id = sched_id
+        """Attach the skill_id of the parent skill
+
+        Args:
+            sched_id (str): skill_id of the parent skill
+        """
+        # NOTE: can not rename sched_id kwarg to keep api compatibility
+        self.skill_id = sched_id
 
     def _create_unique_name(self, name):
         """Return a name unique to this skill using the format
```

Comment:
Maybe clarify that this is a unique `schedule name` so it's tied to the scheduler and not the skill manager?

Reply:
this is the skill_id we are all used to, comes from the parent skill. no idea why sched_id was used but `schedule name` is not a thing